pkg/controller/actuator_reconcile.go

```diff
@@ -144,8 +144,6 @@ func (a *actuator) Reconcile(ctx context.Context, _ logr.Logger, network *extens
 		} else {
 			if ipFamilies.Has(extensionsv1alpha1.IPFamilyIPv6) {
```

Comment:
Maybe, this `if` should also be removed, resulting in `setPoolMode(networkConfig, ipFamilies, calicov1alpha1.Never) for both IP-Families?